python/ray/_raylet.pyx

```diff
@@ -531,6 +538,8 @@ cdef CRayStatus task_execution_handler(
                 # it does, that indicates that there was an internal error.
                 execute_task(task_type, ray_function, c_resources, c_args,
                              c_arg_reference_ids, c_return_ids, returns)
+                import gc;gc.collect()
```

Comment:
This is likely to have a high latency impact, do we need to have it for the final version?

Reply:
I think it was just for debugging.

Reply:
I forgot that this was actually to make one of the tests pass, but @edoakes is working on modifying the test so we don't need it(/figuring out if it's actually a bug).